src/bruno/StringCalculator.ts

```diff
@@ -0,0 +1,51 @@
+export class StringCalculator {
+  add(input: string) {
+    if (this.isEmpty(input)) return 0;
+
+    let delimiterRegex = new RegExp(`\\n|,`, "g");
+    const delimiter = this.findDelimiter(input);
+
+    if (delimiter) {
+      delimiterRegex = new RegExp(`\\n|,|${delimiter}`, "g");
+    }
```

Comment:
Este es solo un tema de estilo de código y es discutible.

Personalmente, creo que evitar las mutaciones ayuda a entender mejor el código y evitar sorpresas. Entonces intento evitar el `let` o mutar objetos siempre que sea posible.

Obviamente es discutible, pero creo que esta es una buena alternativa al let:

```suggestion
    const delimiter = this.findDelimiter(input);

    const delimiterRegex = delimiter ? new RegExp(`\\n|,|${delimiter}`, "g")
      : new RegExp(`\\n|,`, "g");
```

Reply:
Creo que en un principio pensaba hacer mas tareas dentro del bloque if, pero al final quedó una sola línea. Tambien suelo evitar let, a menos que sea necesario. En este caso no lo es, sumo tu sugerencia. Gracias!